middleware/instrument_test.go

```diff
@@ -104,45 +117,89 @@ func NewInstrument(registry *prometheus.Registry) Instrument {
 			Name: "inflight_requests",
 			Help: "Current number of inflight requests.",
 		}, []string{"method", "route"}),
-		RequestCutoff:  SlowRequestCutoff,
+		RequestCutoff:  slowRequestCutoff,
 		ThroughputUnit: throughputUnit,
 		RequestThroughput: reg.NewHistogramVec(prometheus.HistogramOpts{
-			Name:                            "request_throughput_" + throughputUnit,
-			Help:                            "Server throughput running requests.",
-			ConstLabels:                     prometheus.Labels{"cutoff_ms": strconv.FormatInt(SlowRequestCutoff.Milliseconds(), 10)},
-			Buckets:                         []float64{1, 5, 10},
-			NativeHistogramBucketFactor:     metricsNativeHistogramFactor,
-			NativeHistogramMaxBucketNumber:  100,
-			NativeHistogramMinResetDuration: time.Hour,
+			Name:        "request_throughput_" + throughputUnit,
+			Help:        "Server throughput running requests.",
+			ConstLabels: prometheus.Labels{"cutoff_ms": strconv.FormatInt(slowRequestCutoff.Milliseconds(), 10)},
+			Buckets:     []float64{1, 5, 10},
 		}, []string{"method", "route"}),
 	}
 }
 
 func TestExtractValueFromMultiValueHeader(t *testing.T) {
 	tests := []struct {
+		testName string
 		header   string
+		name     string
 		key      string
-		expected int64
+		expected float64
 		err      bool
 	}{
-		{"key0=0, key1=1", "key0", 0, false},
-		{"key0=0, key1=1", "key1", 1, false},
-		{"key0=0, key1=1", "key2", 0, true},
-		{"key0=1.0, key1=1", "key0", 1, false},
-		{"foo", "foo", 0, true},
-		{"foo=bar", "foo", 0, true},
-		{"", "foo", 0, true},
+		{
+			testName: "ExistantKeyInName1",
+			header:   "name0;key0=0.0;key1=1.1, name1;key0=1.1",
+			name:     "name0",
+			key:      "key0",
+			expected: 0.0,
+			err:      false,
+		},
+		{
+			testName: "NonExistantName1",
```

Comment:
did you mean `NonExistantName2`?

Reply:
There are two tests for `NonExistantName`, `NonExistantName1` and `NonExistantName2`

Reply:
i see. maybe it will be less confusing if the keys and the test names used different strings